R/compare_models.R

```diff
@@ -0,0 +1,118 @@
+#' Function for comparing models 
+#' 
+#' Function \code{compare_models} enables user to compare various model with
+#' the same type of task. Function compares them based on metric which can be chosen by the user 
+#' and returns the best model 
+#' 
+#' @param models list of models to compare. Models shoud be either \code{forester_model} or \code{explainer} object.
+#' @param data_test data frame for evaluation models
+#' @param target character indicating the target column in data test table
+#' @param metric string containing name of mertic based on which the model will be selected. 
+#' Metric shoud be written in small letters. Can be omited.
+#'
+#'
+#' @return An object of the class \code{forester_model}.
+#'
+#' @references Explanatory Model Analysis. Explore, Explain, and Examine Predictive Models. \url{https://ema.drwhy.ai/}
+#' @importFrom stats median weighted.mean
+#' @export
+#' @examples
+#' \donttest{
+#' library(DALEX)
+#' data(apartments, package="DALEX")
+#' 
+#' lightgbm_explained <- make_lightgbm(apartments, "m2.price", "regression")
+#' ranger_explained <- make_ranger(apartments, "m2.price", "regression")
+#' 
+#' models <- list(lightgbm_explained, ranger_explained)
+#' 
+#' best_model <- compare_models(models, apartments, "m2.price", "rmse")
+#'}
+##
+
+compare_models <- function(models, data_test, target, metric = NULL){
+  
+  ### Starting conditions 
+  if (class(models) != "list"){
+    stop("Models should be passed as a list")
+  }
+  
+  if (length(models) == 0){
+    stop("List of models is empty.")
+  }
+  
+  # Variable for checking models type 
+  if (!any(class(models[[1]]) %in% c("forester_model", "explainer"))){
+    stop("Wrong class of model. Models should have forester_model or explainer class.")
```

Comment:
Maybe "Wrong class of model. Model should be build with forester function." or something like that.